moveit_ros/planning_interface/test/move_group_interface_cpp_test.cpp

```diff
@@ -0,0 +1,414 @@
+  std::vector<double> plan_joint_positions;
+  move_group_->getCurrentState()->copyJointGroupPositions(
+      move_group_->getCurrentState()->getJointModelGroup(PLANNING_GROUP), plan_joint_positions);
+
+  // Now, let's modify the joint positions.  (radians)
+  ASSERT_EQ(plan_joint_positions.size(), std::size_t(7));
+  plan_joint_positions = { 1.2, -1.0, -0.1, -2.4, 0.0, 1.5, 0.6 };
+  move_group_->setJointValueTarget(plan_joint_positions);
+
+  // plan and move
+  planAndMove();
+
+  // test that we moved to the expected joint positions
+  testJointPositions(plan_joint_positions);
+}
+
+TEST_F(MoveGroupTestFixture, PathConstraintTest)
+{
+  SCOPED_TRACE("PathConstraintTest");
+
+  // set a custom start state
+  geometry_msgs::Pose start_pose;
+  start_pose.orientation.w = 1.0;
+  start_pose.position.x = 0.55;
+  start_pose.position.y = -0.05;
+  start_pose.position.z = 0.8;
+  planAndMoveToPose(start_pose);
+
+  // create an orientation constraint
+  moveit_msgs::OrientationConstraint ocm;
+  ocm.link_name = "panda_link7";
+  ocm.header.frame_id = "panda_link0";
+  ocm.orientation.w = 1.0;
+  ocm.absolute_x_axis_tolerance = 0.1;
+  ocm.absolute_y_axis_tolerance = 0.1;
+  ocm.absolute_z_axis_tolerance = 0.1;
+  ocm.weight = 1.0;
+  moveit_msgs::Constraints test_constraints;
+  test_constraints.orientation_constraints.push_back(ocm);
+  move_group_->setPathConstraints(test_constraints);
+
+  // move to a custom target pose
+  geometry_msgs::Pose target_pose;
+  target_pose.orientation.w = 1.0;
+  target_pose.position.x = 0.28;
+  target_pose.position.y = -0.2;
+  target_pose.position.z = 0.5;
+  planAndMoveToPose(target_pose);
+
+  // clear path constraints
+  move_group_->clearPathConstraints();
+
+  // get the pose after the movement
+  testPose(target_pose);
+}
+
+TEST_F(MoveGroupTestFixture, CartPathTest)
+{
+  SCOPED_TRACE("CartPathTest");
+
+  // set a custom start state
+  geometry_msgs::Pose start_pose;
+  start_pose.orientation.w = 1.0;
+  start_pose.position.x = 0.55;
+  start_pose.position.y = -0.05;
+  start_pose.position.z = 0.8;
+  planAndMoveToPose(start_pose);
+
+  std::vector<geometry_msgs::Pose> waypoints;
+  waypoints.push_back(start_pose);
+
+  geometry_msgs::Pose target_waypoint = start_pose;
+  target_waypoint.position.z -= 0.2;
+  waypoints.push_back(target_waypoint);  // down
+
+  target_waypoint.position.y -= 0.2;
+  waypoints.push_back(target_waypoint);  // right
+
+  target_waypoint.position.z += 0.2;
+  target_waypoint.position.y += 0.2;
+  target_waypoint.position.x -= 0.2;
+  waypoints.push_back(target_waypoint);  // up and left
+
+  moveit_msgs::RobotTrajectory trajectory;
+  const double jump_threshold = 0.0;
+  const double eef_step = 0.01;
+  move_group_->computeCartesianPath(waypoints, eef_step, jump_threshold, trajectory);
+
+  // Execute trajectory
+  moveit::planning_interface::MoveGroupInterface::Plan cartesian_plan;
+  cartesian_plan.trajectory_ = trajectory;
```

Comment:
There is no need for this workaround anymore in `master`. You can just run `move_group_->execute(trajectory)`. If a tutorial (in its `master` version) still specifies this, we should adjust that too.

Reply:
Thank you, here is an issue for tracking the change to the tutorial: https://github.com/ros-planning/moveit_tutorials/issues/479

Reply:
PR: https://github.com/ros-planning/moveit_tutorials/pull/480

Reply:
You are awesome, thanks :sunglasses: